protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java

```diff
@@ -20,6 +20,13 @@
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception
     {
+        // See Varint21FrameDecoder for the general reasoning. We add this here as ByteToMessageDecoder#handlerRemoved()
+        // will fire any cumulated data through the pipeline, so we want to try and stop it here.
+        if ( !ctx.channel().isActive() )
+        {
+            return;
```

Comment:
Why do you skip bytes here elsewhere but not here?

Reply:
`MessageToMessageDecoder` doesn't cumulate the buffers like the `ByteToMessageDecoder` does. We intentionally consume the contents of the message in `ByteToMessageDecoder` as otherwise it would simply save the messages for later, wasting memory for a connection that we know for certain is dead.